broker/src/main/java/io/camunda/zeebe/broker/jobstream/JobGatewayStreamer.java

```diff
@@ -40,9 +78,53 @@ public Optional<GatewayStream<JobActivationProperties, ActivatedJob>> streamFor(
     if (consumers == null || consumers.isEmpty()) {
       final var jobType = BufferUtil.bufferAsString(streamId);
       actor.run(() -> eventService.broadcast(JobStreamTopics.JOB_AVAILABLE.topic(), jobType));
+      return Optional.empty();
     }
 
-    // TODO: for now keep returning empty, but fill out later
-    return Optional.empty();
+    final var target = pickStream(consumers);
+    final var gatewayStream =
+        new JobGatewayStream(
+            target.properties(), new StreamPusher<>(target.id(), this::send, actor::run));
+    return Optional.of(gatewayStream);
+  }
+
+  private StreamConsumer<JobActivationProperties> pickStream(
+      final Set<StreamConsumer<JobActivationProperties>> consumers) {
+    final var targets = new ArrayList<>(consumers);
+    final var index = random.nextInt(consumers.size());
+
+    return targets.get(index);
+  }
+
+  private CompletableFuture<Void> send(final PushStreamRequest request, final MemberId receiver) {
```

Comment:
I kept this here instead of in the `StreamPusher`, since we push to a specific job stream topic - I expect we'll refactor this once we look into adapting the whole thing in a "client/server" protocol kind of thing, but at least we have minimal separation now.